impl/src/main/java/com/okta/sdk/impl/util/OAuth2Utils.java

```diff
@@ -0,0 +1,91 @@
+package com.okta.sdk.impl.util;
+
+import com.okta.sdk.impl.config.ClientConfiguration;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.bouncycastle.util.io.pem.PemObject;
+import org.bouncycastle.util.io.pem.PemReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.UUID;
+
+public class OAuth2Utils {
+    private static final Logger log = LoggerFactory.getLogger(OAuth2Utils.class);
+
+    /**
+     * Create signed JWT string with the supplied {@link ClientConfiguration} details.
+     *
+     * @param {@link ClientConfiguration} object
+     * @return signed JWT string
+     * @throws {@link InvalidKeySpecException}
+     * @throws {@link NoSuchAlgorithmException}
+     * @throws {@link IOException}
+     */
+    public static String createSignedJWT(ClientConfiguration clientConfiguration)
```

Comment:
Turn this into a service instead of a static util class.  That way it can be easily tested (including the code that interacts with it)

Reply:
Got rid of this static util class and moved the helper functions into AccessTokenRetrieverService.